File: GuessTheNumber.py
import random
num = random.randint(0, 100)
guesses=1
guess= input("Guess the number. ")
while guess!=num:
    if guess.isdigit()==True:
        guess=int(guess)
    else:
        print("You had to guess a number. Rerun to play again. ")
        quit()
    if guess>num:
        guesses+=1
        guess = input("Too High! Guess again. ")
    elif guess<num:
        guesses+=1
        guess = input("Too Low! Guess again. ")

print("Correct! The answer was {}. You got it in {} guesses.".format(num, guesses))
    

File: GuessTheNumber.java
import java.util.Random;
import java.util.Scanner;

public class GuessTheNumber {
    public static void main(String[] args) {
        Random rand = new Random();
        Scanner scan = new Scanner(System.in);
        int num = rand.nextInt(100) + 1;
        System.out.println("Guess the number between 1 and 100!");
        int guess;
        int guesses = 0;

        do {
            guess = Integer.parseInt(scan.nextLine());
            guesses++;
            if (guess > num) {
                System.out.println("Your guess is too big!");
            } else if (guess < num) {
                System.out.println("Your guess is too small!");
            }
        } while (guess != num);

        System.out.println(String.format("Congrats! You guessed the number in %s attempts", guesses));
    }
}
